### PR TITLE
Transparent windows

### PR DESCRIPTION
**Pull request by ckaran**
These are the changes that @andsve made sometime ago that I've merged into the latest code of the master branch (commit ecd04539ec9c608d664311b97e8894742912a70c).  This works with OS X 10.9.5 **but it does not yet work on Linux**.  I don't know Linux well enough to be able to make it work with the GLX extensions.  I'm hoping someone else will be able to figure it.

### CODE REVIEW

**Comment by ckaran** [comment]
@andsve Hey, I have no idea if GitHub is doing something funny, but it's claiming you commented on a pull request I just made that occurred **after** your comment.  Anyways, I merged in your changes to the master GLFW branch, and tested it on OS X 10.9.5.  It all works.  Do you have any idea what it will take to make this work under Linux?  I am **not** an OpenGL expert, so I'm kind of just mashing keys while trying to the get the GLX stuff up and running.  If you have any ideas, I'd love to hear about them!

File: src/window.c [hunk 1]
```diff
@@ -135,12 +135,14 @@ GLFWAPI GLFWwindow* glfwCreateWindow(int width, int height,
     fbconfig  = _glfw.hints.framebuffer;
     ctxconfig = _glfw.hints.context;
     wndconfig = _glfw.hints.window;
+    fbconfig.alphaMask  = _glfw.hints.framebuffer.alphaMask ? GLFW_TRUE : GLFW_FALSE;
```
Comment (elmindreda) [inline src/window.c]: Clamping should be done in `glfwWindowHint`.

File: src/window.c [hunk 2]
```diff
@@ -135,12 +135,14 @@ GLFWAPI GLFWwindow* glfwCreateWindow(int width, int height,
...
 
     wndconfig.width   = width;
     wndconfig.height  = height;
     wndconfig.title   = title;
     wndconfig.monitor = (_GLFWmonitor*) monitor;
     ctxconfig.share   = (_GLFWwindow*) share;
+    wndconfig.alphaMask = _glfw.hints.framebuffer.alphaMask ? GLFW_TRUE : GLFW_FALSE;
```
Comment (elmindreda) [inline src/window.c]: Clamping should be done in `glfwWindowHint`.

File: .gitignore [hunk 3]
```diff
@@ -73,3 +73,5 @@ tests/title
 tests/version
 tests/windows
 
+# Local build directory
+build/
```
Comment (elmindreda) [inline .gitignore]: Remove this.

File: include/GLFW/glfw3.h [hunk 4]
```diff
@@ -649,6 +649,7 @@ extern "C" {
 #define GLFW_SRGB_CAPABLE           0x0002100E
 #define GLFW_REFRESH_RATE           0x0002100F
 #define GLFW_DOUBLEBUFFER           0x00021010
+#define GLFW_ALPHA_MASK             0x00021011
```
Comment (elmindreda) [inline include/GLFW/glfw3.h]: Rename to "transparent" throughout.

File: src/window.c (hunk 2 above)
Comment (elmindreda) [inline src/window.c]: This is already assigned above.

File: src/window.c (hunk 1 above)
Comment (elmindreda) [inline src/window.c]: This is already assigned above.

File: .gitignore (hunk 3 above)
Comment (ckaran) [inline .gitignore]: @elmindreda Just fixed this.

**Comment by andsve** [comment]
Whoop, didn't see it was a PR at first. :) I would maybe wanted to clean it up and add Linux before merge, but I like where this is going!

@ckaran No good ideas regarding Linux, but in the original issue thread someone posted a working snippet for Linux, might be a good start.

File: include/GLFW/glfw3.h (hunk 4 above)
Comment (ckaran) [inline include/GLFW/glfw3.h]: @elmindreda Just fixed.  There are two commits, one for `GLFW_ALPHA_MASK` (69fca03d781c125e38d5b6c59448d0496231c3b6), and one for  `alphaMask` (c8c7ad249a682590e80eb383b07771692ce8dbfa)

File: src/window.c (hunk 1 above)
Comment (ckaran) [inline src/window.c]: @elmindreda Fixed in f5e83caeeb09ab2168f907ace3bbc0bf50cc6537.

File: src/window.c (hunk 2 above)
Comment (ckaran) [inline src/window.c]: @elmindreda Fixed in f5e83caeeb09ab2168f907ace3bbc0bf50cc6537.

**Pull request closed without merging**